5.sc_Morphology_Logistic_Regression/2.test_model/scripts/0.test_single_class_model.py

```diff
@@ -0,0 +1,378 @@
+    dmso_holdout = data_all[
+        data_all["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] == control
+    ].sample(n=trt_wells, random_state=seed)
+    # remove the downsampled DMSO wells from the data
+    data_all = data_all
+
+
+# In[11]:
+
+
+# set model path from parameters
+if (aggregation == True) and (nomic == True):
+    model_path = pathlib.Path(
+        f"models/single_class/{cell_type}/aggregated_with_nomic/{MODEL_TYPE}/{control}__{treatment}"
+    )
+elif (aggregation == True) and (nomic == False):
+    model_path = pathlib.Path(
+        f"models/single_class/{cell_type}/aggregated/{MODEL_TYPE}/{control}__{treatment}"
+    )
+elif (aggregation == False) and (nomic == True):
+    model_path = pathlib.Path(
+        f"models/single_class/{cell_type}/sc_with_nomic/{MODEL_TYPE}/{control}__{treatment}"
+    )
+elif (aggregation == False) and (nomic == False):
+    model_path = pathlib.Path(
+        f"models/single_class/{cell_type}/sc/{MODEL_TYPE}/{control}__{treatment}"
+    )
+else:
+    print("Error")
+
+
+# In[12]:
+
+
+model_types = ["final", "shuffled_baseline"]
+feature_types = ["CP"]
+phenotypic_classes = [treatment]
+
+
+# In[13]:
+
+
+# define metadata columns
+# subset each column that contains metadata
+metadata = data_all.filter(regex="Metadata")
+# drop all metadata columns
+data_x = data_all.drop(metadata.columns, axis=1)
+labeled_data = data_all["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"]
+evaluation_types = ["train", "test"]
+
+
+train_labeled_data = data_all.loc[data_all["label"] == "train"][
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+]
+test_labeled_data = data_all.loc[data_all["label"] == "test"][
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+]
+
+
+# In[14]:
+
+
+# set path for figures
+if (aggregation == True) and (nomic == True):
+    figure_path = pathlib.Path(
+        f"./figures/single_class/{cell_type}/aggregated_with_nomic/{control}__{treatment}"
+    )
+    results_path = pathlib.Path(
+        f"./results/single_class/{cell_type}/aggregated_with_nomic/{control}__{treatment}"
+    )
+elif (aggregation == True) and (nomic == False):
+    figure_path = pathlib.Path(
+        f"./figures/single_class/{cell_type}/aggregated/{control}__{treatment}"
+    )
+    results_path = pathlib.Path(
+        f"./results/single_class/{cell_type}/aggregated/{control}__{treatment}"
+    )
+elif (aggregation == False) and (nomic == True):
+    figure_path = pathlib.Path(
+        f"./figures/single_class/{cell_type}/sc_with_nomic/{control}__{treatment}"
+    )
+    results_path = pathlib.Path(
+        f"./results/single_class/{cell_type}/sc_with_nomic/{control}__{treatment}"
+    )
+elif (aggregation == False) and (nomic == False):
+    figure_path = pathlib.Path(
+        f"./figures/single_class/{cell_type}/sc/{control}__{treatment}"
+    )
+    results_path = pathlib.Path(
+        f"./results/single_class/{cell_type}/sc/{control}__{treatment}"
+    )
+else:
```

Comment:
I think there's a bug in your github co-pilot. c: it is providing you the incorrect format for the if statments